numba/typed/typedlist.py

```diff
@@ -373,12 +397,137 @@ def unbox_listtype(typ, val, c):
 
     lstruct.data = builder.load(data_pointer)
     lstruct.meminfo = mi
+    pyapi.decref(miptr)
 
-    lstobj = lstruct._getvalue()
-    c.pyapi.decref(miptr)
 
-    return NativeValue(lstobj)
+def _new_list(context, builder, pyapi, lstruct, itemty, size, error_handler):
+    """ Initialize new typed list from struct during unboxing. """
+    # create ptr to new typed list
+    ptr = listobject._list_new_codegen(
+        context, builder, itemty, size, error_handler)
+    lstruct.data = ptr
+    # setup method table
+    listobject._list_codegen_set_method_table(
+        context, builder, ptr, itemty)
+    # setup builder representation for typed-list
+    listobject._add_meminfo(context, builder, lstruct)
+
+
+def _convert_python_list_to_numba_typed_list(
```

Comment:
This segfault is coming from here:
https://github.com/numba/numba/pull/4883/files/ea1683ca4bb127f9a0479a15432a0a8e8d87fa6a#diff-ae60f5afa43090234f80b19d5450963fR530
because the signalling bit isn't propagated, so an Exception state is entered but the "this wasn't a successful conversion, abort" signalling bit is missing. Try changing that line to:
```python
return NativeValue(lstobj, is_error=c.builder.load(errorptr))
```

Reply:
Yes, that did the trick and fixed the issue. Thank you! 🙏

Reply:
no problem.